src/types/Typechecker/Util.hs

```diff
@@ -285,13 +285,17 @@ subtypeOf ty1 ty2
     | isStackboundType ty1 =
         liftM (isStackboundType ty2 &&) $ unbox ty1 `subtypeOf` unbox ty2
     | isArrowType ty1 && isArrowType ty2 = do
-        let argTys1 = getArgTypes ty1
-            argTys2 = getArgTypes ty2
+        let typeParams1 = getTypeParameters ty1
```

Comment:
I agree (`sub` and `super` are good names I think), but suggest that I do this in a different refactoring PR so that anyone looking at the history will find a clean change for this problem. OK?